WMIAdventure/backend/WMIAdventure_backend/battle/businesslogic/BattleCard.py

```diff
@@ -49,3 +74,34 @@ def assign_buff(self, buff: Buff, effect_type: Union[CardEffect.EffectId, None]
             card_type = effect.effect_model.card_effect.id
             if effect_type is None or card_type == effect_type:
                 effect.add_buff(buff)
+
+    def __update_states__(self):
+        """
+        Updates card's states and removes not active ones.
+        :return: None.
+        """
+
+        for state in self.states:
+            state.update()
+
+        self.__remove_inactive_states__()
+
+    def __update__(self):
+        """
+        Updates card by updating effects and states.
+        :return: None.
+        """
+
+        self.__update_effects__()
+        self.__update_states__()
+
+    def __remove_inactive_states__(self):
+        """
+        Removes states that are not active.
+        :return: None.
+        """
+
+        state: CardState
+        for state in self.states:
+            if not state.active:
+                self.states.remove(state)
```

Comment:
Nie rozumiem tej listy stanów, po co ona jest? Karta może być we więcej niż jednym stanie na raz?

Reply:
Faktycznie, powinna być kolejka stanów karty, a karta powinna być w jednym stanie naraz